test/operators/prunejob.go

```diff
@@ -0,0 +1,94 @@
+package operators
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/osde2e/pkg/helper"
+
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+)
+
+var _ = ginkgo.Describe("[OSD] Prune jobs", func() {
+	h := helper.New()
+	ginkgo.Context("pruner jobs should works", func() {
+		namespace := "openshift-sre-pruning"
+		cronJobs := []string{"builds-pruner", "deployments-pruner", "image-pruner"}
+		for _, cronJob := range cronJobs {
+			ginkgo.It(cronJob+" should run successfully", func() {
+				getOpts := metav1.GetOptions{}
+				cjob, err := h.Kube().BatchV1beta1().CronJobs(namespace).Get(cronJob, getOpts)
+				Expect(err).NotTo(HaveOccurred())
+				job := createJobFromCronJob(cjob)
+				job, err = h.Kube().BatchV1().Jobs(namespace).Create(job)
+				defer func() {
+					err = h.Kube().BatchV1().Jobs(namespace).Delete(job.Name, &metav1.DeleteOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				}()
+				Expect(err).NotTo(HaveOccurred())
+				err = waitJobComplete(h, namespace, job.Name)
+				Expect(err).NotTo(HaveOccurred())
+			})
```

Comment:
Can you also set a timeout for the individual test? See https://github.com/openshift/osde2e/blob/master/test/operators/dedicatedadmin.go#L75 for an example.

Reply:
@jeefy Updated

Reply:
Thanks!